### ISSUE TITLE
FSK Creator GUI issue with the sheet combo box

### ISSUE THREAD

**Issue by miguelalba** [SiLeBAT/FSK-Lab#260]
The combo box with the list of sheets should be populated when the spreadsheet is assigned. Currently it only works when a local path (full path) is used for the spreadsheet. It does not work for KNIME urls.

|  ![knime_mountpoint](https://user-images.githubusercontent.com/19906968/46353023-76aa5c80-c65b-11e8-899c-e8590687826b.PNG) | ![local_path](https://user-images.githubusercontent.com/19906968/46353024-76aa5c80-c65b-11e8-88d5-24676ea34f54.PNG) |
| :-: | :-: |
| mount point | full path |

**Comment by miguelalba** [comment]
The dialog of the creator node fails to load paths with the knime URL like in the example: *knime://knime.workflow/data/metadata.xlsx*. Yet the node model is using the same code to import the metadata flawlessly.

```
java.io.IOException: No context for relative URL available
	at org.knime.workbench.explorer.ExplorerURLStreamHandler.resolveKNIMEURL(ExplorerURLStreamHandler.java:155)
	at org.knime.workbench.explorer.ExplorerURLStreamHandler.openConnection(ExplorerURLStreamHandler.java:110)
	at org.eclipse.osgi.internal.url.URLStreamHandlerProxy.openConnection(URLStreamHandlerProxy.java:114)
	at java.net.URL.openConnection(URL.java:979)
	at org.knime.core.util.FileUtil.resolveToPath(FileUtil.java:1135)
	at de.bund.bfr.knime.fsklab.nodes.CreatorNodeDialog$SheetFieldTask.doInBackground(CreatorNodeDialog.java:265)
	at de.bund.bfr.knime.fsklab.nodes.CreatorNodeDialog$SheetFieldTask.doInBackground(CreatorNodeDialog.java:1)
	at javax.swing.SwingWorker$1.call(SwingWorker.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at javax.swing.SwingWorker.run(SwingWorker.java:334)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by miguelalba** [comment]
**This is not a bug.**

The knime mountpoint ARE supported (D:\FSKLabWorkspace\Listeria_model_QMRA\data\metadata.xlsx) but not the keyword *knime.workflow*. The latter one only works for flow variables

**Comment by llavall** [comment]
Sorry, but I don't see any solution here? We need to use the workflow option to define Models inside a workflow to be send around. Only the sheet function does not work with it and this needs to be fixed - maybe you find a workaround or you just remove the selection option? I think there is a sheet selection in the excel nodes from KNIME - may be this gives you new ideas?

**Comment by miguelalba** [comment]
I opened this issue as I thought the KNIME urls were not working and later learned it is only the *knime.workflow* keyword that is not supported in the dialogs.

The KNIME methods to resolve the paths like `FileUtils.resolveToPath` are not supporting this keyword, therefore it does not work in current dialogs of the FSK-Lab nodes. It is working though when used in flow variables and this is implemented by KNIME. I ignore whether the dialogs of other KNIME nodes do support it. If they do then it makes sense to keep this issue open.

Another unrelated issue is the sheet combo box which is automatically populated with the list of sheets when a spreadsheet is selected. Unfortunately, the KNIME dialogs ignores the values set through flow variables. Thus when the spreadsheet is set by a flow variable, the sheet combo box only sees the empty spreadsheet field and remains empty, hence the need to set the sheet through a flow variable too. I ignore whether any node is able to react to values set by flow variables and have not found any yet. This should be treated as a separate issue.

**Comment by miguelalba** [comment]
Fixed in [v1.2.2](https://github.com/SiLeBAT/FSK-Lab/commit/dfb0dc4cbb5f602839c6aab8d307e08618f2caa8)
